### ISSUE TITLE
`XDG_BIN_HOME` common default not used

### ISSUE THREAD

**Issue by thernstig** [astral-sh/uv#6937]
In case someone has not set `XDG_BIN_HOME`, it installs into `~/.cargo/bin/` on Linux.

A common convention is however to install into `$HOME/.local/bin`. Example: https://gist.github.com/roalcantara/107ba66dfa3b9d023ac9329e639bc58c

The benefit is of course the same as with anything else for the XDG spec. It keeps the home dir clutter free.

**Comment by charliermarsh** [comment]
I think we intend to change the install directory, but it's something of a breaking change so may come in a minor release.

**Comment by thernstig** [comment]
I am trying to introduce uv into an organization, due to that it obviously is far superior to anything else for anyone who has followed the Python echo system for years. Though I know some developers will frown about creation of directories in the root of the home dir, so this change is, believe it or not, something I will want to wait for. Some devs can be pretty adamant about minor things 😄

**Comment by zanieb** [comment]
We did this a while back.

**Comment by HaleTom** [comment]
For googlers:

`$XDG_BIN_HOME` is NOT part of the standard:
https://specifications.freedesktop.org/basedir/latest/

It does say:

> User-specific executable files may be stored in $HOME/.local/bin. Distributions should ensure this directory shows up in the UNIX $PATH environment variable, at an appropriate place.

> Since $HOME might be shared between systems of different architectures, installing compiled binaries to $HOME/.local/bin could cause problems when used on systems of differing architectures. This is often not a problem, but the fact that $HOME becomes partially architecture-specific if compiled binaries are placed in it should be kept in mind.

**Comment by thernstig** [comment]
True that, but that said, ~/.local/bin is widely used as a de facto location for user executables, and many tools and docs refer to it even though it is not standardized by the spec

**Comment by HaleTom** [comment]
@thernstig we are in vehement agreement :)

Re-quoting from the spec:
> User-specific executable files may be stored in `$HOME/.local/bin`

### TL;DR:  It is the variable `$XDG_BIN_HOME` that is NOT standard.  `$HOME/.local/bin` IS standard.
